From: Thomas Renninger <trenn@suse.de>
Subject: ACPI: Store valid ACPI tables passed via early initrd in reserved memblock areas
References: none
Patch-Mainline: submitted

A later patch will compare them with ACPI tables that get loaded at boot or
runtime and if criteria match, a stored one is loaded.

Signed-off-by: Thomas Renninger <trenn@suse.de>
CC: lenb@kernel.org
CC: robert.moore@intel.com
CC: yinghai@kernel.org
CC: hpa@zytor.com
CC: eric.piel@tremplin-utc.net

Automatically created from "patches.arch/acpi_override_2_2.patch" by xen-port-patches.py

--- head.orig/arch/x86/kernel/setup-xen.c	2012-08-20 14:37:06.000000000 +0200
+++ head/arch/x86/kernel/setup-xen.c	2012-10-04 14:24:35.000000000 +0200
@@ -1067,6 +1067,11 @@ void __init setup_arch(char **cmdline_p)
 
 	reserve_initrd();
 
+#ifdef CONFIG_ACPI
+	if (is_initial_xendomain())
+		acpi_initrd_override((void *)initrd_start, initrd_end - initrd_start);
+#endif
+
 #ifndef CONFIG_XEN
 	reserve_crashkernel();
 
